### PR TITLE
Add support for generic SPI NOR flash and the GD25Q128 on Rock64

### PR DESCRIPTION
**Pull request by xalius**
Rock64 has a GD25Q128 SPI NOR Flash device, add support for mtd devices and generic NOR drivers.

### CODE REVIEW

File: arch/arm64/boot/dts/rockchip/rk3328.dtsi [hunk 1]
```diff
@@ -461,12 +461,12 @@
...
-		dmas = <&dmac 8>, <&dmac 9>;
-		#dma-cells = <2>;
-		dma-names = "tx", "rx";
+		/* dmas = <&dmac 8>, <&dmac 9>; */
+		/* #dma-cells = <2>; */
+		/* dma-names = "tx", "rx"; */
 		pinctrl-names = "default";
 		pinctrl-0 = <&spi0m2_clk &spi0m2_tx &spi0m2_rx &spi0m2_cs0>;
-		status = "disabled";
+		status = "okay";
```
Comment (ayufan) [inline arch/arm64/boot/dts/rockchip/rk3328.dtsi]: Please enable this in `rock64.dts`.

```
&spi0 {
  status = "okay";
};

File: arch/arm64/boot/dts/rockchip/rk3328.dtsi [hunk 2]
```diff
@@ -461,12 +461,12 @@
 		#size-cells = <0>;
 		clocks = <&cru SCLK_SPI>, <&cru PCLK_SPI>;
 		clock-names = "spiclk", "apb_pclk";
-		dmas = <&dmac 8>, <&dmac 9>;
-		#dma-cells = <2>;
-		dma-names = "tx", "rx";
+		/* dmas = <&dmac 8>, <&dmac 9>; */
```
Comment (ayufan) [inline arch/arm64/boot/dts/rockchip/rk3328.dtsi]: I believe that you should be able to remove these keys too from rock64.dts. Or maybe we should just "zero" them?

File: arch/arm64/configs/rockchip_linux_defconfig [hunk 3]
```diff
@@ -776,7 +788,6 @@ CONFIG_USB_EHCI_HCD_PLATFORM=y
 CONFIG_USB_OHCI_HCD=y
 # CONFIG_USB_OHCI_HCD_PCI is not set
 CONFIG_USB_OHCI_HCD_PLATFORM=y
-CONFIG_USB_ACM=m
```
Comment (ayufan) [inline arch/arm64/configs/rockchip_linux_defconfig]: Why we removed that?

**Review by ayufan — CHANGES_REQUESTED** [review]

**Comment by xalius** [comment]
Cleaned that up a little, I don't know where the two stray config changes came from, but they are gone now :-), removing dt properties via /delete-properties/  works, I haven't found a nicer way to disable DMA...

**Review by ayufan — APPROVED** [review]

**Pull request merged**